### PR TITLE
GUVNOR-2338: Copy/Rename modal doesn't disappear when used from Project Explorer links

### PR DESCRIPTION
**Pull request by paulovmr**
Depends on: https://github.com/uberfire/uberfire-extensions/pull/174

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java [hunk 1]
```diff
@@ -326,22 +329,36 @@ public void execute( final String comment ) {
...
         popup.show();
     }
 
+    protected DataModelerErrorCallback getCopyErrorCallback( final CopyPopupView copyPopupView ) {
+        return new DataModelerErrorCallback( Constants.INSTANCE.modelEditor_copying_error() ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java]: For consistency with the ```show()``` we should call ```CopyPopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java [hunk 2]
```diff
@@ -472,14 +489,15 @@ private boolean isSourceDirty() {
         return originalSourceHash != null && originalSourceHash != getSource().hashCode();
     }
 
-    private RemoteCallback<Path> getCopySuccessCallback() {
+    private RemoteCallback<Path> getCopySuccessCallback( final CopyPopupView copyPopupView ) {
         return new RemoteCallback<Path>() {
 
             @Override
             public void callback( final Path response ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java]: For consistency with the ```show()``` we should call ```CopyPopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java [hunk 3]
```diff
@@ -841,18 +861,31 @@ private void rename( final boolean saveCurrentChanges ) {
...
         popup.show();
     }
 
+    protected DataModelerErrorCallback getRenameErrorCallback( final RenamePopupView renamePopupView ) {
+        return new DataModelerErrorCallback( Constants.INSTANCE.modelEditor_renaming_error() ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                renamePopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/DataModelerScreenPresenter.java]: For consistency with the ```show()``` we should call ```RenamePopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java [hunk 4]
```diff
@@ -620,25 +623,39 @@ public void execute( final FileNameAndCommitMessage details ) {
...
         };
     }
 
+    protected HasBusyIndicatorDefaultErrorCallback getCopyErrorCallback( final CopyPopupView copyPopupView ) {
+        return new HasBusyIndicatorDefaultErrorCallback( busyIndicatorView ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java]: For consistency with the ```show()``` we should call ```CopyPopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java [hunk 5]
```diff
@@ -649,22 +666,35 @@ public void execute( final FileNameAndCommitMessage details ) {
...
         };
     }
 
+    protected HasBusyIndicatorDefaultErrorCallback getRenameErrorCallback( final RenamePopupView renamePopupView ) {
+        return new HasBusyIndicatorDefaultErrorCallback( busyIndicatorView ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                renamePopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java]: For consistency with the ```show()``` we should call ```RenamePopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/menu/ProjectMenu.java [hunk 6]
```diff
@@ -106,16 +126,32 @@ public void execute( FileNameAndCommitMessage payload ) {
...
                                          @Override
                                          public void execute( FileNameAndCommitMessage payload ) {
-                                             renameService.call().rename( path,
-                                                                          payload.getNewFileName(),
-                                                                          payload.getCommitMessage() );
+                                             renameService.call( new RemoteCallback<Void>() {
+
+                                                 @Override
+                                                 public void callback( final Void o ) {
+                                                     renamePopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/menu/ProjectMenu.java]: For consistency with the ```show()``` we should call ```RenamePopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/menu/ProjectMenu.java [hunk 7]
```diff
@@ -106,16 +126,32 @@ public void execute( FileNameAndCommitMessage payload ) {
...
+                                                 }
+                                             }, new ErrorCallback<Void>() {
+
+                                                 @Override
+                                                 public boolean error( final Void o,
+                                                                       final Throwable throwable ) {
+                                                     renamePopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/menu/ProjectMenu.java]: For consistency with the ```show()``` we should call ```RenamePopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java [hunk 8]
```diff
@@ -257,25 +263,48 @@ public void callback( Object response ) {
...
+                             },
+                             renamePopupView
                            );
     }
 
+    protected RemoteCallback<Void> getRenameSuccessCallback( final RenamePopupView renamePopupView ) {
+        return new RemoteCallback<Void>() {
+            @Override
+            public void callback( final Void o ) {
+                renamePopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: For consistency with the ```show()``` we should call ```RenamePopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java [hunk 9]
```diff
@@ -257,25 +263,48 @@ public void callback( Object response ) {
...
+        };
+    }
+
+    protected HasBusyIndicatorDefaultErrorCallback getRenameErrorCallback( final RenamePopupView renamePopupView ) {
+        return new HasBusyIndicatorDefaultErrorCallback( baseView ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                renamePopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: For consistency with the ```show()``` we should call ```RenamePopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java [hunk 10]
```diff
@@ -298,22 +327,44 @@ public void callback( Object response ) {
...
+                           },
+                           copyPopupView
                          );
     }
 
+    protected RemoteCallback<Void> getCopySuccessCallback( final CopyPopupView copyPopupView ) {
+        return new RemoteCallback<Void>() {
+            @Override
+            public void callback( final Void o ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: For consistency with the ```show()``` we should call ```CopyPopup.hide()```.

File: kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java [hunk 11]
```diff
@@ -298,22 +327,44 @@ public void callback( Object response ) {
...
+        };
+    }
+
+    protected HasBusyIndicatorDefaultErrorCallback getCopyErrorCallback( final CopyPopupView copyPopupView ) {
+        return new HasBusyIndicatorDefaultErrorCallback( baseView ) {
+
+            @Override
+            public boolean error( final Message message,
+                                  final Throwable throwable ) {
+                copyPopupView.hide();
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-project-explorer/kie-wb-common-project-explorer-client/src/main/java/org/kie/workbench/common/screens/explorer/client/widgets/BaseViewPresenter.java]: For consistency with the ```show()``` we should call ```CopyPopup.hide()```.

**Comment by manstis** [comment]
Other than the change to use the Presenter's ```hide()``` method I think this is much clearer to follow than the flag you had before. Thanks for making the changes... once you switch to using the Presenter I'll OK the merge.

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
master: c60f46b576bde63f0d1afbcdb7fb30f373189d48

**Pull request closed without merging**